src/mon/MDSMonitor.cc

```diff
@@ -1190,6 +1190,26 @@ bool MDSMonitor::preprocess_command(MonOpRequestRef op)
       }
     }
     r = 0;
+  } else if (prefix == "fs feature ls") {
+    // fw to leader?
+    if (is_leader()) {
```

Comment:
i don't really understand this part. if `fs feature ls` is a read-only command, shall we handle right here by peon as well? or shall we read this feature list from an agreed repository m like mds map? if the only reliable source of information, we should forward this command to leader instead of being quiet if it hits a peon.

Reply:
grep asked me to do this. saving the feature list in mdsmap is tricky too( how do update the list when a new feature is added?)

Reply:
see https://github.com/ceph/ceph/pull/31726#discussion_r376302952 . i think OSD addresses this problem using require_osd_release .

Reply:
there is 'ceph fs set <name> min_compat_client xx' command, which is similar to require_osd_release. The problem is that it require client to support all features introduce by given release. kernel client may only support some features.

Reply:
I'm not very fussed about this read-only output, but the prior way the code worked it *did* forward to the leader if received on a peon — you have to set ```r != -1``` for it to be handled locally (the end: label interprets -1 as "not handled" and returns false to the preprocess caller, which forces a forward), which IIRC this only did for the leader where it dumped the output.